### PR TITLE
matrix-gtk-theme: init at 0-unstable-2024-07-22

### PR DESCRIPTION
**Pull request by D3vil0p3r**
## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by kirillrdy** [comment]
doesn't build

```
/nix/store/z004gzhs1p6z755lrq4qzvzrp9b0r8xb-stdenv-linux/setup: line 1652: /nix/store/7yl49vbb62620lkl7zwicyzvls1jnmsz-matrix-gtk-theme-0-unstable-2024-07-15/share/icons/: No such file or directory

```

**Comment by D3vil0p3r** [comment]
> doesn't build
> 
> ```
> /nix/store/z004gzhs1p6z755lrq4qzvzrp9b0r8xb-stdenv-linux/setup: line 1652: /nix/store/7yl49vbb62620lkl7zwicyzvls1jnmsz-matrix-gtk-theme-0-unstable-2024-07-15/share/icons/: No such file or directory
> ```

Which nix-build arguments are you using to build it?

**Comment by D3vil0p3r** [comment]
> doesn't build
> 
> ```
> /nix/store/z004gzhs1p6z755lrq4qzvzrp9b0r8xb-stdenv-linux/setup: line 1652: /nix/store/7yl49vbb62620lkl7zwicyzvls1jnmsz-matrix-gtk-theme-0-unstable-2024-07-15/share/icons/: No such file or directory
> ```

@kirillrdy sorry, now it should be fixed.

File: pkgs/by-name/ma/matrix-gtk-theme/package.nix [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      ${lib.optionalString (tweakVersions != [ ]) "--tweaks " + toString tweakVersions} \
+      -d "$out/share/themes"
+      cd ../icons
+      ${lib.optionalString (iconVariants != [ ]) ''
+        mkdir -p $out/share/icons
+        cp -a ${toString (map (v: "Matrix-${v}") iconVariants)} $out/share/icons/
+      ''}
+      runHook postInstall
+    '';
+
```
Comment (JohnRTitor) [inline pkgs/by-name/ma/matrix-gtk-theme/package.nix:112]:
```suggestion
  passthru.updateScript = unstableGitUpdater { };

```

File: pkgs/by-name/ma/matrix-gtk-theme/package.nix [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    meta = with lib; {
```
Comment (JohnRTitor) [inline pkgs/by-name/ma/matrix-gtk-theme/package.nix:113]: remove `with lib;` here and prepend the needed components with `lib.` as appropiate

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 327431` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>matrix-gtk-theme</li>
  </ul>
</details>

File: pkgs/by-name/ma/matrix-gtk-theme/package.nix (hunk 1 above)
Comment (D3vil0p3r) [reply pkgs/by-name/ma/matrix-gtk-theme/package.nix:112]: @JohnRTitor For what is used it?

File: pkgs/by-name/ma/matrix-gtk-theme/package.nix (hunk 2 above)
Comment (D3vil0p3r) [reply pkgs/by-name/ma/matrix-gtk-theme/package.nix]: @JohnRTitor fixed.

Comment (D3vil0p3r) [reply pkgs/by-name/ma/matrix-gtk-theme/package.nix]: But no idea why the GH check gives us unformatted.

File: pkgs/by-name/ma/matrix-gtk-theme/package.nix (hunk 1 above)
Comment (JohnRTitor) [reply pkgs/by-name/ma/matrix-gtk-theme/package.nix:115]: This is a update script that automatically fetches the latest commit and updates the hash for hassle free update procedure.

```bash
nix-shell maintainers/scripts/update.nix --argstr package matrix-gtk-theme --argstr commit true
```
using it like this should automatically commit the new revision.

**Review by JohnRTitor — APPROVED** [review]
LGTM

**Pull request merged**